techtv2ovs/utils.py

```diff
@@ -0,0 +1,314 @@
+                    'private_token': token
+                }
+            )
+            s3videos = get_s3_videos(external_id)
+            if not s3videos:
+                ttvvideo.status = ImportStatus.MISSING
+                ttvvideo.videofile_status = ImportStatus.MISSING
+                ttvvideo.save()
+                continue
+            if not ttvvideo.video:
+                ttvvideo.video = Video.objects.create(
+                    collection=ttvcollection.collection,
+                    title=v_title,
+                    description=remove_tags(v_description),
+                    source_url="http://techtv.mit.edu/videos/",
+                    is_private=ttvvideo.private,
+                )
+            ttvvideo.status = ImportStatus.CREATED
+            ttvvideo.save()
+            self.process_files(ttvvideo, s3videos)
+
+    def process_collection(self, cid, name, description, email):
+        """
+        Iterate over every TechTV collection
+
+        Args:
+            cid (int): TechTV collection ID
+            name (str): TechTV collection name
+            description (str): TechTV collection description
+            email (str): TechTV collection owner's email address
+        """
+        ttvcollection, _ = TechTVCollection.objects.get_or_create(id=cid, defaults={
+            'name': name,
+            'description': description,
+            'owner_email': email
+        })
+        self.output.write('Processing collection: {}'.format(name))
+        if not ttvcollection.collection:
+            view_list, _ = MoiraList.objects.get_or_create(name=moiralist_name(cid, name))
+            admin_list, _ = MoiraList.objects.get_or_create(name='{}-owner'.format(view_list.name))
+            ovs_collection = Collection.objects.create(
+                title=name,
+                description=remove_tags(description),
+                owner=get_owner(email),
+            )
+            ovs_collection.admin_lists = [admin_list]
+            ovs_collection.view_lists = [view_list]
+            ovs_collection.save()
+            ttvcollection.collection = ovs_collection
+            ttvcollection.save()
+        self.process_videos(ttvcollection)
+        ttvcollection.status = ImportStatus.COMPLETE
+
+    def extract_collections(self):
+        """
+        Yield TechTV collections from MySQL
+
+        Yields:
+            tuple: Tuple of database rows (as tuples) from the collections table
+
+        """
+        collection_query = (
+            'SELECT DISTINCT c.id, c.name, c.description, u.email FROM collections c '
+            'INNER JOIN collection_videos cv ON c.id = cv.collection_id '
+            'INNER JOIN videos v on v.id = cv.video_id '
+            'LEFT JOIN users u ON c.user_id = u.id '
+            'WHERE v.status = \'approved\' and v.file_restored = 1 AND v.locked = 0 AND c.protected = %s'
+        )
+        if self.collection:
+            collection_query += " and c.id = {}".format(self.collection)
```

Comment:
Can you change this to use a parameter instead of string formatting?

Reply:
✅